### PR TITLE
Change names of pipe operators following issue #178

### PR DESCRIPTION
**Pull request by prosoitos**
**Edited 2018-10-17**

Replace:

- magrittr forward-pipe operator
- magrittr compound assignment pipe-operator
- magrittr exposition pipe-operator
- magrittr tee operator

With:

- pipe
- assignment pipe
- exposition pipe
- tee pipe

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ argument, e.g. `f <- . %>% abs %>% mean(na.rm = TRUE)`.
 Three new operators are introduced for some special cases
 
 * Compound assignment pipe: `%<>%`
-* Tee pipe: `%T>%`
+* Tee assignment pipe: `%T>%`
```
Comment (hadley) [inline NEWS.md]: This should just be tee

File: R/pipe.R [hunk 2]
```diff
@@ -188,15 +188,15 @@ pipe <- function()
 #' @export
 `%>%`  <- pipe()
 
-#' magrittr compound assignment pipe-operator
+#' compound assignment pipe
```
Comment (hadley) [inline R/pipe.R]: Should start sentence with capital letter

Comment (prosoitos) [reply R/pipe.R]: Will capitalize them all (did the same for all 4 pipes). I was trying to respect the original, but I wasn't sure if the lower case had been deliberate or a way not to change the package name.

File: NEWS.md (hunk 1 above)
Comment (prosoitos) [reply NEWS.md]: Right. I had some confusion as to the use of the word "assignment" as you can see in the conversation of issue #178. Following that conversation, I was wondering if the names:
```
pipe
assignment pipe
exposition pipe
tee pipe
```
might not be better???

Comment (hadley) [reply NEWS.md]: Sure, sounds good.

File: R/pipe.R (hunk 2 above)
Comment (hadley) [reply R/pipe.R]: magrittr is a proper name that is not capitalised, so it doesn't get capitalised even when at the start of a sentence.

Comment (prosoitos) [reply R/pipe.R]: Right

File: NEWS.md (hunk 1 above)
Comment (prosoitos) [reply NEWS.md]: Ok. Will update them all to that then.

Comment (prosoitos) [reply NEWS.md]: Done

File: R/pipe.R (hunk 2 above)
Comment (prosoitos) [reply R/pipe.R]: Corrected

File: R/is_something.R [hunk 3]
```diff
@@ -19,30 +19,30 @@ is_parenthesized <- function(expr)
   is.call(expr) && identical(expr[[1L]], quote(`(`))
 }
 
-# Check whether a pipe is a tee.
+# Check whether a pipe is the tee pipe operator.
 #
 # @param pipe A (quoted) pipe
 # @return logical - TRUE if pipe is a tee, FALSE otherwise.
-is_tee <- function(pipe)
+is_tee_pipe <- function(pipe)
```
Comment (hadley) [inline R/is_something.R]: Can you please revert the changes to function names? That would be a breaking change, and I don't think it's worth the cost.

File: R/is_something.R [hunk 4]
```diff
@@ -19,30 +19,30 @@ is_parenthesized <- function(expr)
...
 # @param pipe A (quoted) pipe
 # @return logical - TRUE if pipe is the dollar pipe, FALSE otherwise.
-is_dollar <- function(pipe)
+is_exposition_pipe <- function(pipe)
 {
   identical(pipe, quote(`%$%`))
 }
 
-# Check whether a pipe is the compound assignment pipe operator
+# Check whether a pipe is the assignment pipe operator.
```
Comment (hadley) [inline R/is_something.R]: Aren't we removing "operator"?

File: R/is_something.R (hunk 3 above)
Comment (prosoitos) [reply R/is_something.R]: OK

Note: are they used in other repositories? I made sure to change them in the entire repository. So I don't think that I broke anything.

Comment (hadley) [reply R/is_something.R]: Any changes to functions require that we fully understand the implications to other packages.

Comment (prosoitos) [reply R/is_something.R]: OK. Will revert then.

File: R/is_something.R (hunk 4 above)
Comment (prosoitos) [reply R/is_something.R]: Yes. Sorry. I thought those were descriptive sentences. But it makes more sense to use the new names indeed. Will change.

File: R/is_something.R (hunk 3 above)
Comment (prosoitos) [reply R/is_something.R]: Done.

File: R/is_something.R (hunk 4 above)
Comment (prosoitos) [reply R/is_something.R]: Done.
